src/client/dfs/duns.c

```diff
@@ -162,15 +162,15 @@ duns_resolve_lustre_path(const char *path, struct duns_attr_t *attr)
 		D_ERROR("unable to allocate XATTR_SIZE_MAX to get LOV/LMV "
 			"for '%s', errno %d(%s).\n", path, errno,
 			strerror(errno));
-		/** TODO - convert errno to rc */
-		return -DER_NOSPACE;
+		return ENOMEM;
 	}
 	fd = open(path, O_RDONLY | O_DIRECTORY | O_NOFOLLOW);
 	if (fd == -1 && errno != ENOTDIR) {
+		int err = errno;
```

Comment:
Why do you need err variable ?

Reply:
Correct error handling of errno is really difficult, D_ERROR() will almost certainly update errno to a different value.

Reply:
Hmm ok, but seems that this save of original errno is already done in core logging routine d_vlog() !

Reply:
We have been doing this in other instances too. so it's not just this.
I have not looked into our error stack to check all the possibilities, but is d_vlog() the only underlying routine called? any thing like printf can change errno, so i think this was safer.
unless you are sure, then i can change it..

Reply:
No I am fine with this (BTW, I have already approved this PR!), I was only asking because I thought it should already be handled inside D_ERROR() as it is a very generic macro. But I think only Cart people could tell if errno  protection is really implemented or not.